python/lsst/meas/modelfit/truncatedGaussian.cc

```diff
@@ -0,0 +1,137 @@
+
+using Sampler = TruncatedGaussianSampler;
+using Evaluator = TruncatedGaussianEvaluator;
+using LogEvaluator = TruncatedGaussianLogEvaluator;
+
+using PyTruncatedGaussian = py::class_<TruncatedGaussian, std::shared_ptr<TruncatedGaussian>>;
+using PySampler = py::class_<Sampler, std::shared_ptr<Sampler>>;
+using PyEvaluator = py::class_<Evaluator, std::shared_ptr<Evaluator>>;
+using PyLogEvaluator = py::class_<LogEvaluator, std::shared_ptr<LogEvaluator>>;
+
+template <typename Class, typename PyClass>
+PyClass declareEvaluator(py::module & mod, std::string const & name) {
+    PyClass cls(mod, ("TruncatedGaussian" + name).c_str());
+    cls.def(py::init<TruncatedGaussian const &>(), "parent"_a);
+    cls.def(
+        "__call__",
+        (Scalar (Class::*)(ndarray::Array<Scalar const,1,1> const &) const) &Class::operator(),
+        "alpha"_a
+    );
+    cls.def(
+        "__call__",
+        (void (Class::*)(
+            ndarray::Array<Scalar const,2,1> const &,
+            ndarray::Array<Scalar,1,1> const &
+        ) const) &Class::operator(),
+        "alpha"_a, "output"_a
+    );
+    return cls;
+}
+
+PYBIND11_PLUGIN(truncatedGaussian) {
+
+    py::module mod("truncatedGaussian");
+
+    if (_import_array() < 0) {
+        PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+        return nullptr;
+    }
+
+    py::module::import("lsst.afw.math");
+
+    PyTruncatedGaussian cls(mod, "TruncatedGaussian");
+    py::enum_<TruncatedGaussian::SampleStrategy>(cls, "SampleStrategy")
+        .value("DIRECT_WITH_REJECTION", TruncatedGaussian::DIRECT_WITH_REJECTION)
+        .value("ALIGN_AND_WEIGHT", TruncatedGaussian::ALIGN_AND_WEIGHT)
+        .export_values();
+    cls.def_static("fromSeriesParameters", &TruncatedGaussian::fromSeriesParameters,
+                   "q0"_a, "gradient"_a, "hessian"_a);
+    cls.def_static("fromStandardParameters", &TruncatedGaussian::fromStandardParameters,
+                   "mean"_a, "covariance"_a);
+    cls.def(
+        "sample",
+        (Sampler (TruncatedGaussian::*)(TruncatedGaussian::SampleStrategy) const) &TruncatedGaussian::sample,
+        "strategy"_a
+    );
+    cls.def(
+        "sample",
+        (Sampler (TruncatedGaussian::*)(Scalar) const) &TruncatedGaussian::sample,
+        "minRejectionEfficiency"_a=0.1
+    );
+    cls.def("evaluateLog", &TruncatedGaussian::evaluateLog);
+    cls.def("evaluate", &TruncatedGaussian::evaluate);
+    cls.def("getDim", &TruncatedGaussian::getDim);
+    cls.def("maximize", &TruncatedGaussian::maximize);
+    cls.def("getUntruncatedFraction", &TruncatedGaussian::getUntruncatedFraction);
+    cls.def("getLogPeakAmplitude", &TruncatedGaussian::getLogPeakAmplitude);
+    cls.def("getLogIntegral", &TruncatedGaussian::getLogIntegral);
+
+    cls.attr("LogEvaluator") = declareEvaluator<LogEvaluator, PyLogEvaluator>(mod, "LogEvaluator");
+    cls.attr("Evaluator") = declareEvaluator<Evaluator, PyEvaluator>(mod, "Evaluator");
```

Comment:
Why not `cls.attr("Sampler") =...`? Please add a code comment explaining why this is missing, or, perhaps, why the above two are present.